lib/restAPIs/library.js

```diff
@@ -0,0 +1,105 @@
+  _getBase() {
+    return 'library';
+  }
+
+  /**
+   * Return object representing the library for the authenticated user.
+   * @param {array[String]} outputFlag - Filtered output. DEAFULT = include everything, FILTER_TOC = filter out cahpters and pages.
+   * @param {String} fields
+   */
+  getLibrary(outputFlag = 'DEFAULT', fields = '') {
+    const queryParameter = buildCommonQuery(fields);
+    queryParameter.outputFlag = outputFlag;
+
+    return this._makeRequest(
+      `${this._getBase()}`,
+      queryParameter,
+      'GET',
+      null,
+      null
+    ).then((result) => this.throwIfFailed(result, 200));
+  }
+
+  /**
+   * Publish an object (document or dossier) in project defined by projectID.
+   * Information about object to publish and recipient are detailed in object 'body'.
+   * https://demo.microstrategy.com/MicroStrategyLibrary/api-docs/index.html#/Library/publishObject
+   * @param {String} projectID -
+   * @param {JSON object} body
+   */
+  publishObject(projectID, body) {
+    const endpoint = `${this._getBase()}`;
+    const customHeaders = this.getProjectHeader(projectID);
+    return this._makeRequest(
+      endpoint,
+      body,
+      'POST',
+      customHeaders
+    ).then((result) => this.throwIfFailed(result, 204));
+  }
+
+  /**
+   * Return object (document or dossier) identified by 'objectId' from project identified by 'projectId'
+   * @param {String} projectId
+   * @param {String} documentId
+   * @param {String} fields - Fields to be included in the result. i.e. 'id,elements'
+   */
+  getObject(projectId, objectId, fields = '') {
+    const queryParameter = buildCommonQuery(fields);
+    const endpoint = `${this._getBase()}/${objectId}`;
+    const customHeaders = this.getProjectHeader(projectId);
+    return this._makeRequest(
+      endpoint,
+      queryParameter,
+      'GET',
+      customHeaders
+    ).then((result) => this.throwIfFailed(result, 200));
+  }
+
+  /**
+   * Delete an object (dossier or document) identified by 'objectId' from project identified by 'projectId'
+   * @param {Strin} projectId
+   * @param {String} objectId
+   */
+  deleteObject(projectId, objectId) {
+    const endpoint = `${this._getBase()}/${objectId}`;
+    const queryParameter = buildCommonQuery();
+    const customHeaders = this.getProjectHeader(projectId);
+
+    return this._makeRequest(
+      endpoint,
+      queryParameter,
+      'DELETE',
+      customHeaders
+    ).then((result) => this.throwIfFailed(result, 204));
+  }
+
+  /**
+   * Delete object (dossier or document) identified by 'objectId' from project identified by 'projectId' of user identified by 'userId'
+   * @param {String} projectId
+   * @param {Strin} objectId
+   * @param {String} userId
+   */
+  deleteUserObject(projectId, objectId, userId) {
+    const endpoint = `${this._getBase()}/${objectId}/recipients/${userId}`;
+    const queryParameter = buildCommonQuery();
+    const customHeaders = this.getProjectHeader(projectId);
+
```

Comment:
If any params are required, can you leverage the assertParam or assert functions? See the other wrappers for examples. These functions through errors if the key in object is undefined, or the object itself is undefined, respectively.

Reply:
Done.